sklearn/linear_model/_ridge.py

```diff
@@ -2212,16 +2264,22 @@ def fit(self, X, y, sample_weight=None):
             parameters = {"alpha": alphas}
             solver = "sparse_cg" if sparse.issparse(X) else "auto"
             model = RidgeClassifier if is_classifier(self) else Ridge
+            est = model(
+                fit_intercept=self.fit_intercept,
+                solver=solver,
+            )
+            if _routing_enabled():
+                est.set_fit_request(sample_weight=True)
+
             gs = GridSearchCV(
-                model(
-                    fit_intercept=self.fit_intercept,
-                    solver=solver,
-                ),
+                est,
                 parameters,
                 cv=cv,
                 scoring=self.scoring,
             )
-            gs.fit(X, y, sample_weight=sample_weight)
+            if sample_weight is not None:
+                params["sample_weight"] = sample_weight
```

Comment:
I would have expected this from the start whether we use a grid-search or the GCV estimator.